zingolib/src/wallet/traits.rs

```diff
@@ -599,6 +603,10 @@ impl DomainWalletExt for SaplingDomain {
         &tree_state.sapling_tree
     }
 
+    fn protocol() -> ShieldedProtocol {
+        ShieldedProtocol::Sapling
```

Comment:
Given that this trait is being implemented on a domain that has a type including Domain specitic information..   why aren't the trait methods referencing something via (Self)?   Maybe the same thing for each implementation..  maybe even a default implementation?